pyshelf/routes/artifact.py

```diff
@@ -107,3 +107,33 @@ def delete_metadata_property(container, bucket_name, path, item):
         response = response_map.map_metadata_result_errors(result)
 
     return response
+
+
+@artifact.route("/<bucket_name>/artifact/_search", methods=["POST"])
+@decorators.foundation
+def root_search(container, bucket_name):
+    response = search(container)
+    return response
+
+
+@artifact.route("/<bucket_name>/artifact/<path:path>/_search", methods=["POST"])
+@decorators.foundation
+def path_search(container, bucket_name, path):
+    response = search(container)
+    return response
+
+
+def search(container):
+    """
+        Does a search with the given criteria.
+
+        Args:
+            container(pyshelf.container.Container)
+
+        Returns:
+            Flask response
+    """
+    criteria = json.loads(request.data)
```

Comment:
it looks like the request should have a get_json(force=True) method so json.loads doesn't need to be used directly.  The force is required in case the mimetype wasn't sent as application/json.  I have a feeling most people calling this api will not send that header.